cli/tools/task.rs

```diff
@@ -60,6 +60,11 @@ pub async fn execute_script(
         .await;
     Ok(exit_code)
   } else if let Some(script) = package_json_scripts.get(task_name) {
+    ps.package_json_deps_installer
+      .ensure_top_level_install()
+      .await?;
+    ps.npm_resolver.resolve_pending().await?;
```

Comment:
Wouldn't it make sense for `ensure_top_level_install` to call `npm_resolver.resolve_pending` internally?

Reply:
No because when resolving with deno_graph we may call `ensure_top_level_install`, then wait for the full graph to be resolved, then finally call `resolve_pending()`. This is the only place they're done together so maybe in the future we could consider a helper method, but a helper method does make it more likely to accidentally call both. Also, the way the structs are right now with the lsp would require a bit of a big refactor to make this work. The code isn't great at the moment and more refactoring in the future would be nice.

Reply:
Thanks for the explanation